client/src/components/formula-attributes-view/Assignment.js

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import TextField from '@mui/material/TextField';
+
+const Assignment = ({ assignment, attributes, handleAttributeChange, assignmentIndex }) => {
+
+  const attributeTextFields = () => {
+    return (
+      attributes.map((attribute, attributeIndex) => (
+        <TextField
+          type='text'
+          key={attribute}
+          variant='standard' 
+          label={attribute}
+          InputLabelProps={{ shrink: true }}
+          margin='normal'
+          sx={{
+            marginTop: 0,
+            width: '100%'
+          }}
+          onChange={event => handleAttributeChange(assignmentIndex, attributeIndex, event)}
+        />
+      ))
+    );
+  };
+
+  return (
+    <Box sx={{
+      bgcolor: '#FFFFFF',
+      display: 'flex',
+      flexDirection: 'column',
+      boxShadow: 2, 
+      borderRadius: 2,
+      px: 3,
+      py: 1,
```

Comment:
I would have liked a bit more padding on the bottom so I would change this to: `pt: 1, pb: 1.5`. But I guess this depends on what you like.